src/commands/publish.js

```diff
@@ -0,0 +1,210 @@
+          process.exit(0)
+        }
+        let versionIncrement
+        if (!flags.version) {
+          versionIncrement = Object.keys(flags)[0].toUpperCase()
+        }
+        const bodyPublishExtensionVersion = {
+          dynamicComponentFileId: dynamicComponentFileActivated.id,
+          version: flags.version,
+          versionIncrement
+        }
+        try {
+          await this.publishExtensionVersion(bodyPublishExtensionVersion, token)
+        } catch (error) {
+          if (error.response.status === 422) {
+            this.error('Intended version is less than or equal to last version')
+          } else {
+            this.error(error.response)
+          }
+          process.exit(0)
+        }
+        this.success('New version is published with success')
+      }
+    } catch (error) {
+      console.log(chalk.red(`${error}`))
+    }
+  }
+  async confirmVersion (version, date) {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    })
+    return new Promise((resolve, reject) => {
+      rl.question(`Publish version ${chalk.blue(version)} created at ${chalk.blue(moment(date).format('LLLL'))} on the marketplace? Yes/No `, answer => {
+        rl.close()
+        if (
+          answer.toLowerCase() === 's' ||
+            answer.toLowerCase() === 'sim' ||
+            answer.toLowerCase() === 'yes' ||
+            answer.toLowerCase() === 'y'
+        ) {
+          resolve(version)
+        } else {
+          console.log(chalk.red('operation canceled'))
+          resolve(false)
+        }
+      })
+    })
+  }
+  commandSintaxeValid (flags) {
+    return Object.keys(flags).length < 2
+  }
+  async publishExtensionVersion (body, token) {
+    await api.axios.post(
+      `/${credentials.institution}/marketplace/extensions/publish-version`,
+      body,
+      { headers: { Authorization: `Bearer ${token}` } }
+    )
+  }
+  async publishExtension (body, token) {
+    await api.axios.post(
+      `/${credentials.institution}/marketplace/extensions`,
+      body,
+      { headers: { Authorization: `Bearer ${token}` } }
+    )
+  }
+  async getResponseFromUser (requestText, responseDefault) {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    })
+    return new Promise(resolve => {
+      rl.question(requestText, answer => {
+        rl.close()
+        if (answer) { resolve(answer) } else resolve(responseDefault)
+      })
+    })
+  }
+  async confirmQuestion (requestText, responseText) {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    })
+    return new Promise((resolve, reject) => {
+      rl.question(requestText, answer => {
+        rl.close()
+        if (
+          answer.toLowerCase() === 's' ||
+            answer.toLowerCase() === 'sim' ||
+            answer.toLowerCase() === 'yes' ||
+            answer.toLowerCase() === 'y'
+        ) {
```

Comment:
Eu escreveria dessa forma:

`['s', 'sim', 'yes', 'y'].includes(answer.toLowerCase())`

E para melhorar, visto que isso deve ser recorrente no Quoti CLI eu teria criado uma função simples genérica:

```javascript
String.prototype.isYes = function () {
    return ['s', 'sim', 'yes', 'y'].includes(this.toLowerCase())
}
```

Dessa forma no if você poderia simplesmente fazer:

`answer.isYes()`

Reply:
@ovictoraurelio, tem um alguns downsides ao permitir que o prototype seja mudado:
* Desenvolvedores mais leigos podem sobreescrever funções padrão e causar erros difíceis de achar. 
* Além disso, podem se acostumar com uma função que existe em um projeto e causar erros em outros em que a função não existe.
* Outros dois motivos aqui (artigo curto, 30s de leitura): https://flaviocopes.com/javascript-why-not-modify-object-prototype/

Para ser justo, nenhuma das razões que eu dei ou vi se aplicam a esse caso mas o problema é a prática de fazer isso, eu considero uma má prática no geral e não traz nenum benefício gritante em comparação a essa alternativa:
Ao invés de fazer isso, poderiamos simplesmente seguir o nosso padrão de utilizar um Util e pronto, o que já acontece em todos os projetos da empresa.

Reply:
Agreed! Gostei da resposta bem detalhada. Já é útil pra colocar no Developer Guide da gente.

Reply:
![](https://media3.giphy.com/media/3o7abB06u9bNzA8lu8/giphy.gif)

Reply:
done